avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -206,14 +207,17 @@ public static boolean isSupportedForFastSerializer(Schema.Type schemaType) {
     FastDeserializer<?> deserializer = fastSpecificRecordDeserializersCache.get(schemaKey);
 
     if (deserializer == null) {
-      deserializer = fastSpecificRecordDeserializersCache.putIfAbsent(schemaKey,
-          new FastDeserializerWithAvroSpecificImpl<>(writerSchema, readerSchema));
-      if (deserializer == null) {
-        deserializer = fastSpecificRecordDeserializersCache.get(schemaKey);
+      AtomicBoolean status = new AtomicBoolean(false);
+      deserializer = fastSpecificRecordDeserializersCache.computeIfAbsent(
+          schemaKey,
```

Comment:
This seems too complicate.
Shall we just move the following logic into the `computeIfAbsent`?
```
CompletableFuture.supplyAsync(() -> buildSpecificDeserializer(writerSchema, readerSchema), executor)
            .thenAccept(d -> fastSpecificRecordDeserializersCache.put(schemaKey, d));
```

Reply:
`d -> fastSpecificRecordDeserializersCache.put(schemaKey, d)` is inside computeIfAbsent, this update will fail due to nested update not allowed exception.